sdk/component/runner.go

```diff
@@ -0,0 +1,143 @@
+package component
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/signal"
+	"syscall"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type (
+	// runner is used to setup the run context.
+	runner struct {
+		config *RunnerConfig
+	}
+
+	// componentRunnerFunc is an alias for executing component run functions.
+	componentRunnerFunc func(context.Context, *RunnerConfig) error
+)
+
+// newRunner returns a new runner which is initialised looking at the default configuration
+// and can be customised applying options.
+// Returns an error if the configuration is invalid.
+func newRunner(opts ...RunnerOption) (*runner, error) {
```

Comment:
We create a new runner by introspecting the environment variables, then customising it (if options are passed) and then finally validating it for required bits to be well formed.